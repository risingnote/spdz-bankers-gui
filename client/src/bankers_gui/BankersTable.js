/**
 * Display the diners table with names of diners from props and whose paying if any.
 */
import React, { PropTypes } from 'react'
import ClassNames from 'classnames'

import './BankersTable.css'

const BankersTable = (props) => {

  const getDinerName = (position => {
    return (position < props.diners.length ? props.diners[position].name : '')
  })

  const placeClass = (position => {
    return ClassNames({
      'place': true,
      'paying': (position < props.diners.length && props.diners[position].paying)
    })
  })

  return (
    <div className='BankersTable'>
      <div className={placeClass(0)} style={{transform: 'translate(0rem, -8.2rem) rotate(180deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-180deg)'}}>{getDinerName(0)}</div>
        </div>
      </div>

      <div className={placeClass(4)} style={{transform: 'translate(5.8rem, -5.8rem) rotate(225deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-225deg)'}}>{getDinerName(4)}</div>
        </div>
      </div>

      <div className={placeClass(2)} style={{transform: 'translate(8.2rem, 0rem) rotate(270deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-270deg)'}}>{getDinerName(2)}</div>
        </div>
      </div>

      <div className={placeClass(5)} style={{transform: 'translate(5.8rem, 5.8rem) rotate(315deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-315deg)'}}>{getDinerName(5)}</div>
        </div>
      </div>

      <div className={placeClass(1)} style={{transform: 'translate(0rem, 8.2rem) rotate(0deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-0deg)'}}>{getDinerName(1)}</div>
        </div>
      </div>

      <div className={placeClass(6)} style={{transform: 'translate(-5.8rem, 5.8rem) rotate(45deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-45deg)'}}>{getDinerName(6)}</div>
        </div>
      </div>

      <div className={placeClass(3)} style={{transform: 'translate(-8.2rem, 0rem) rotate(90deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-90deg)'}}>{getDinerName(3)}</div>
        </div>
      </div>

      <div className={placeClass(7)} style={{transform: 'translate(-5.8rem, -5.8rem) rotate(135deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-135deg)'}}>{getDinerName(7)}</div>
        </div>
      </div>

    </div> 
  )
}

BankersTable.propTypes = {
  diners: PropTypes.array.isRequired
}

export default BankersTable
