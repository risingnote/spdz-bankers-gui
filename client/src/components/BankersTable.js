/**
 * Display the diners table with names of diners from props and whose paying if any.
 * Maxium of 8 places, not all have to be filled. Any more will be ignored.
 */
import React, { PropTypes } from 'react'
import ClassNames from 'classnames'

import './BankersTable.css'

const BankersTable = (props) => {

  const getDinerName = (position => {
    return (position < props.diners.length ? props.diners[position].name : '')
  })

  const placeClass = (position => {
    return ClassNames({
      'place': true,
      'paying': (position < props.diners.length && props.winningClientId !== undefined &&
                 props.diners[position].publicKey === props.winningClientId)
    })
  })

  return (
    <div className='BankersTable'>
      <div className={placeClass(0)} style={{transform: 'translate(0rem, -120px) rotate(180deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-180deg)'}}>{getDinerName(0)}</div>
        </div>
      </div>

      <div className={placeClass(4)} style={{transform: 'translate(85px, -85px) rotate(225deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-225deg)'}}>{getDinerName(4)}</div>
        </div>
      </div>

      <div className={placeClass(2)} style={{transform: 'translate(120px, 0px) rotate(270deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-270deg)'}}>{getDinerName(2)}</div>
        </div>
      </div>

      <div className={placeClass(5)} style={{transform: 'translate(85px, 85px) rotate(315deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-315deg)'}}>{getDinerName(5)}</div>
        </div>
      </div>

      <div className={placeClass(1)} style={{transform: 'translate(0px, 120px) rotate(0deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-0deg)'}}>{getDinerName(1)}</div>
        </div>
      </div>

      <div className={placeClass(6)} style={{transform: 'translate(-85px, 85px) rotate(45deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-45deg)'}}>{getDinerName(6)}</div>
        </div>
      </div>

      <div className={placeClass(3)} style={{transform: 'translate(-120px, 0px) rotate(90deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-90deg)'}}>{getDinerName(3)}</div>
        </div>
      </div>

      <div className={placeClass(7)} style={{transform: 'translate(-85px, -85px) rotate(135deg)'}}>
        <div className='place-setting'>
          <div style={{transform: 'rotate(-135deg)'}}>{getDinerName(7)}</div>
        </div>
      </div>

    </div> 
  )
}

BankersTable.propTypes = {
  diners: PropTypes.array.isRequired,
  winningClientId: PropTypes.string
}

export default BankersTable
