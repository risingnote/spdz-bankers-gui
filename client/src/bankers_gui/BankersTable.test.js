import React from 'react'
import renderer from 'react-test-renderer'

import BankersTable from './BankersTable'

describe('Bankers Table display component rendering', () => {
  it('Renders as expected (compared to a snapshot) 3 diners, none paying', () => {

    const tree = renderer.create(
      <BankersTable diners={[{name: 'me'}, {name: 'Alice'}, {name: 'Bob'}]}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Renders as expected (compared to a snapshot) 5 diners, 1 paying', () => {

    const tree = renderer.create(
      <BankersTable diners={[{name: 'me'}, {name: 'Alice'}, {name: 'Bob'}, {name: 'Mal'}, {name: 'Rich', paying: 'true'}]}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

})
