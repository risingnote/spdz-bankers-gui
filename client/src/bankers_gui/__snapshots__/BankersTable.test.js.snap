exports[`Bankers Table display component rendering Renders as expected (compared to a snapshot) 3 diners, none paying 1`] = `
<div
  className="BankersTable">
  <div
    className="place"
    style={
      Object {
        "transform": "translate(0rem, -120px) rotate(180deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-180deg)",
          }
        }>
        me
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(85px, -85px) rotate(225deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-225deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(120px, 0px) rotate(270deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-270deg)",
          }
        }>
        Bob
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(85px, 85px) rotate(315deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-315deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(0px, 120px) rotate(0deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-0deg)",
          }
        }>
        Alice
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-85px, 85px) rotate(45deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-45deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-120px, 0px) rotate(90deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-90deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-85px, -85px) rotate(135deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-135deg)",
          }
        }>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Bankers Table display component rendering Renders as expected (compared to a snapshot) 5 diners, 1 paying 1`] = `
<div
  className="BankersTable">
  <div
    className="place"
    style={
      Object {
        "transform": "translate(0rem, -120px) rotate(180deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-180deg)",
          }
        }>
        me
      </div>
    </div>
  </div>
  <div
    className="place paying"
    style={
      Object {
        "transform": "translate(85px, -85px) rotate(225deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-225deg)",
          }
        }>
        Rich
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(120px, 0px) rotate(270deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-270deg)",
          }
        }>
        Bob
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(85px, 85px) rotate(315deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-315deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(0px, 120px) rotate(0deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-0deg)",
          }
        }>
        Alice
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-85px, 85px) rotate(45deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-45deg)",
          }
        }>
        
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-120px, 0px) rotate(90deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-90deg)",
          }
        }>
        Mal
      </div>
    </div>
  </div>
  <div
    className="place"
    style={
      Object {
        "transform": "translate(-85px, -85px) rotate(135deg)",
      }
    }>
    <div
      className="place-setting">
      <div
        style={
          Object {
            "transform": "rotate(-135deg)",
          }
        }>
        
      </div>
    </div>
  </div>
</div>
`;
